src/server/pkg/obj/amazon_client.go

```diff
@@ -291,9 +291,11 @@ func (c *amazonClient) Reader(ctx context.Context, name string, offset uint64, s
 		}
 		req.Header.Add("Range", byteRange)
 
-		backoff.RetryNotify(func() error {
-			span, _ := tracing.AddSpanToAnyExisting(ctx, "aws/cloudfront.Get")
-			defer tracing.FinishAnySpan(span)
+		backoff.RetryNotify(func() (retErr error) {
```

Comment:
Should we add these for all storage providers consistently? (ie, GCS, local, etc)

Reply:
That's my goal (to try and handle all of the storage clients equivalently). The Amazon client is slightly unique in that it has the special handling for the case where a cloudfront distribution has been set up in front of S3, and that's what this trace is supposed to capture. Thus, this particular kind of tracing won't be in other clients, but other kinds of tracing should be:

- clients where `Exists` is implemented as some kind of `_, err := realclient.Stat(path); return err == nil` I tried to make sure to add the error to the trace
- clients that have their own implementation of `Reader` and `Writer`, I tried to make sure to instrument the custom calls to `Read` and `Write`